### PR TITLE
SONAR-8798 beta upgrade to es 5

### PR DESCRIPTION
**Pull request by sns-seb**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/es/DefaultIndexSettings.java [hunk 1]
```diff
@@ -43,17 +43,22 @@
 
   public static final String TYPE = "type";
   public static final String INDEX = "index";
-  public static final String ANALYZED = "analyzed";
-  public static final String STRING = "string";
+  public static final String INDEX_SEARCHABLE_FOR_TEXT = "analyzed"; // ES 5: remove this constant and replace usages with INDEX_SEARCHABLE
+  public static final String INDEX_SEARCHABLE_FOR_KEYWORD = "not_analyzed"; // ES 5: change to true and rename to INDEX_SEARCHABLE
+  public static final String INDEX_NOT_SEARCHABLE = "no"; // ES 5: change to false
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/es/DefaultIndexSettings.java]: MINOR Move this trailing comment on the previous empty line. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ATrailingCommentCheck)

File: server/sonar-server/src/main/java/org/sonar/server/es/DefaultIndexSettings.java [hunk 2]
```diff
@@ -43,17 +43,22 @@
...
+  public static final String FIELD_TYPE_TEXT = "string";  // ES 5: change to text
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/es/DefaultIndexSettings.java]: MINOR Move this trailing comment on the previous empty line. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ATrailingCommentCheck)

File: server/sonar-server/src/main/java/org/sonar/server/es/DefaultIndexSettings.java [hunk 3]
```diff
@@ -43,17 +43,22 @@
...
+  public static final String FIELD_TYPE_KEYWORD = "string";  // ES 5: change to keyword
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/es/DefaultIndexSettings.java]: MINOR Move this trailing comment on the previous empty line. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ATrailingCommentCheck)

File: server/sonar-server/src/main/java/org/sonar/server/es/DefaultIndexSettings.java [hunk 4]
```diff
@@ -43,17 +43,22 @@
...
   public static final String STANDARD = "standard";
   public static final String PATTERN = "pattern";
   public static final String CUSTOM = "custom";
   public static final String KEYWORD = "keyword";
   public static final String CLASSIC = "classic";
+  public static final boolean REFRESH_IMMEDIATE = true; // ES 5: replace with org.elasticsearch.action.support.WriteRequest.RefreshPolicy.IMMEDIATE
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/es/DefaultIndexSettings.java]: MINOR Move this trailing comment on the previous empty line. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ATrailingCommentCheck)

File: server/sonar-server/src/main/java/org/sonar/server/es/DefaultIndexSettings.java [hunk 5]
```diff
@@ -43,17 +43,22 @@
...
+  public static final boolean REFRESH_NONE = false; // ES 5: replace with org.elasticsearch.action.support.WriteRequest.RefreshPolicy.NONE
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/es/DefaultIndexSettings.java]: MINOR Move this trailing comment on the previous empty line. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ATrailingCommentCheck)

File: server/sonar-server/src/main/java/org/sonar/server/es/DefaultIndexSettings.java [hunk 6]
```diff
@@ -43,17 +43,22 @@
 
   public static final String TYPE = "type";
   public static final String INDEX = "index";
-  public static final String ANALYZED = "analyzed";
-  public static final String STRING = "string";
+  public static final String INDEX_SEARCHABLE_FOR_TEXT = "analyzed"; // ES 5: remove this constant and replace usages with INDEX_SEARCHABLE
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/es/DefaultIndexSettings.java]: MINOR Move this trailing comment on the previous empty line. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ATrailingCommentCheck)

File: server/sonar-server/src/main/java/org/sonar/server/es/DefaultIndexSettings.java [hunk 7]
```diff
@@ -43,17 +43,22 @@
...
+  public static final String INDEX_SEARCHABLE_FOR_KEYWORD = "not_analyzed"; // ES 5: change to true and rename to INDEX_SEARCHABLE
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/es/DefaultIndexSettings.java]: MINOR Move this trailing comment on the previous empty line. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ATrailingCommentCheck)

File: server/sonar-server/src/main/java/org/sonar/server/es/NewIndex.java [hunk 8]
```diff
@@ -258,9 +267,11 @@ public NewIndexType build() {
       Map<String, Object> hash = new TreeMap<>();
       if (subFields.isEmpty()) {
         hash.putAll(ImmutableMap.of(
-          "type", "string",
-          "index", disableSearch ? "no" : "not_analyzed",
-          "norms", ImmutableMap.of("enabled", String.valueOf(!disableNorms))));
+            "type", FIELD_TYPE_KEYWORD,
+            "index", disableSearch ? INDEX_NOT_SEARCHABLE : INDEX_SEARCHABLE_FOR_KEYWORD,
+            "norms",
+            ImmutableMap.of("enabled", String.valueOf(!disableNorms)) // ES 5: replace with String.valueOf(!disableNorms)
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/es/NewIndex.java]: MINOR Move this trailing comment on the previous empty line. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ATrailingCommentCheck)

File: server/sonar-server/src/main/java/org/sonar/server/es/NewIndex.java [hunk 9]
```diff
@@ -271,18 +282,20 @@ public NewIndexType build() {
...
         multiFields.put(fieldName, ImmutableMap.of(
-          "type", "string",
-          "index", "not_analyzed",
-          "term_vector", termVectorWithPositionOffsets ? "with_positions_offsets" : "no",
-          "norms", ImmutableMap.of("enabled", "false")));
+            "type", FIELD_TYPE_KEYWORD,
+            "index", INDEX_SEARCHABLE_FOR_KEYWORD,
+            "term_vector", termVectorWithPositionOffsets ? "with_positions_offsets" : "no",
+            "norms",
+            ImmutableMap.of("enabled", "false") // ES 5: replace with "false"
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/es/NewIndex.java]: MINOR Move this trailing comment on the previous empty line. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ATrailingCommentCheck)

File: server/sonar-server/src/main/java/org/sonar/server/es/metadata/MetadataIndex.java [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+      return Optional.of(value);
+    }
+    return Optional.empty();
+  }
+
+  private void setMetadata(String hash, String id) {
+    esClient.prepareIndex(MetadataIndexDefinition.INDEX_TYPE_METADATA)
+      .setId(id)
+      .setSource(MetadataIndexDefinition.FIELD_VALUE, hash)
+      .setRefresh(REFRESH_IMMEDIATE) // ES 5: change to setRefreshPolicy
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/es/metadata/MetadataIndex.java]: MINOR Move this trailing comment on the previous empty line. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ATrailingCommentCheck)

File: server/sonar-server/src/main/java/org/sonar/server/permission/index/PermissionIndexer.java [hunk 11]
```diff
@@ -127,7 +128,7 @@ public void deleteProject(String projectUuid) {
     authorizationScopes.forEach(scope -> esClient
       .prepareDelete(scope.getIndexType(), projectUuid)
       .setRouting(projectUuid)
-      .setRefresh(true)
+      .setRefresh(REFRESH_IMMEDIATE) // ES 5: change to setRefreshPolicy
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/permission/index/PermissionIndexer.java]: MINOR Move this trailing comment on the previous empty line. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ATrailingCommentCheck)

File: server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueIndexer.java [hunk 12]
```diff
@@ -143,7 +144,7 @@ public void deleteByKeys(String projectUuid, List<String> issueKeys) {
     BulkRequestBuilder builder = esClient.prepareBulk();
     for (String issueKey : issueKeys) {
       builder.add(esClient.prepareDelete(INDEX_TYPE_ISSUE, issueKey)
-        .setRefresh(false)
+        .setRefresh(REFRESH_NONE) // ES 5: change to setRefreshPolicy
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueIndexer.java]: MINOR Move this trailing comment on the previous empty line. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ATrailingCommentCheck)

File: server/sonar-server/src/main/java/org/sonar/server/measure/index/ProjectMeasuresIndexer.java [hunk 13]
```diff
@@ -90,7 +91,7 @@ public void deleteProject(String uuid) {
     esClient
       .prepareDelete(INDEX_TYPE_PROJECT_MEASURES, uuid)
       .setRouting(uuid)
-      .setRefresh(true)
+      .setRefresh(REFRESH_IMMEDIATE) // ES 5: change to setRefreshPolicy
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/measure/index/ProjectMeasuresIndexer.java]: MINOR Move this trailing comment on the previous empty line. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ATrailingCommentCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 13 issues
* ![MINOR][MINOR] 13 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by bartfastiel** [comment]
These changes will be covered by #2300 .

**Pull request closed without merging**